telebot/async_telebot.py

```diff
@@ -558,7 +558,7 @@ async def _run_middlewares_and_handlers(self, message, handlers, middlewares, up
                             logger.error("It is not allowed to pass data and values inside data to the handler. Check your handler: {}".format(handler['function']))
                             return
                     else:
-                        data_copy = data.copy()
+                        data_copy = copy.deepcopy(data)
```

Comment:
You should no copy data. It is required to pass original objects retrieved from middlewares to handlers